### ISSUE TITLE
MacOS maximized on start detection

### ISSUE THREAD

**Issue by Sheepux** [manga-download/haruneko#23]
Following #7 on macOS the maximized state is not detected at start.

**Comment by sisco0** [comment]
Are we using the methods shown in https://svelte.dev/tutorial/custom-stores such as `set`?

**Comment by Sheepux** [comment]
No, we're using nwjs maximize function that should handle such case by itself.
So this is a nwjs issue, but it probably means that something is missing for MacOS (an app id?)

**Comment by sisco0** [comment]
I attach some lines of interest
https://github.com/manga-download/haruneko/blob/b55df8986f793766d5467bb2797460f22e8e2346/src/frontend/classic/Stores.ts#L4
https://github.com/manga-download/haruneko/blob/a0641d44c30e90230a82937728e79f53d4a86666/src/frontend/classic/Frontend.ts#L12
https://github.com/manga-download/haruneko/blob/57322181ba91c19e09cde9b479c02877e1885e7f/src/frontend/classic/components/AppBar.svelte#L40
https://github.com/manga-download/haruneko/blob/57322181ba91c19e09cde9b479c02877e1885e7f/src/frontend/classic/components/AppBar.svelte#L23-L27
Maybe `winMaximized` is not set to `true` for the _MacOS_ case as this last complex condition is not met as expected?

**Comment by Sheepux** [comment]
winMaximized is only here for the button action change
See https://github.com/manga-download/haruneko/blob/57322181ba91c19e09cde9b479c02877e1885e7f/src/frontend/classic/components/AppBar.svelte#L64

**Comment by Sheepux** [comment]
on windows winMaximized is struggling when moving the window from a 1080p to a 4K screen. Could be related

**Comment by Sheepux** [comment]
Possible issue with equality as the pixel ratio is not the same. There could be an error marging
https://github.com/manga-download/haruneko/blob/4c192674670c282af56e5958b5a1e00f48276cb8/web/src/frontend/classic/components/AppBar.svelte#L36
(still investigating)

**Comment by Pierre-Monier** [comment]
For me (macOS M1 2020) I have no problem with maximized on start detection (on dev and prod), the issue I have is that `winMaximized` is not updated correctly. So I can only `maximize` or `restore` the app once, after that I'm blocked. The code is here : 

```typescript
    function updateWindowState() {
        winMaximized =
            window.outerWidth >= screen.availWidth &&
            window.outerHeight >= screen.availHeight &&
            window.screenTop <= screen['availTop'] &&
            window.screenLeft <= screen['availLeft'];
    }
```

The problem is that the `window` object is not updated inside the `resize` event listener, so `winMaximized` is never updated. I try getting the `window` object from the `UIEvent` but it's the same issue. 

The only way to fix is to use `nw.Window.get()`. But since we wrap this inside the `IWindowController`, the clean way is to update this interface. So I prefer to ask first before I made any change. 

Can you use the maximize button more than once on other platforms ?
